src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioWorkspaceStatusServiceFactory.cs

```diff
@@ -47,79 +49,61 @@ public IWorkspaceService CreateService(HostWorkspaceServices workspaceServices)
         /// </summary>
         private class Service : IWorkspaceStatusService
         {
-            private readonly VisualStudioWorkspace _workspace;
-
-            // only needed for testing
-            private ResettableDelay _lastTimeCalled;
+            private readonly IAsyncServiceProvider2 _serviceProvider;
 
             public event EventHandler<bool> StatusChanged;
 
-            public Service(VisualStudioWorkspace workspace)
+            public Service(IAsyncServiceProvider2 serviceProvider)
             {
-                // until we get new platform API, use legacy one that is not fully do what we want
-                _workspace = workspace;
-            }
+                _serviceProvider = serviceProvider;
 
-            public async System.Threading.Tasks.Task WaitUntilFullyLoadedAsync(CancellationToken cancellationToken)
-            {
-                if (_workspace.Options.GetOption(ExperimentationOptions.SolutionStatusService_ForceDelay))
+                // TODO: there is no right place to register event?
+                Task.Run(RegisterEventAsync, CancellationToken.None);
+
+                async Task RegisterEventAsync()
```

Comment:
could this just be inlined into the Task.Run call.  it seems less clear what's going on in this form to me.  thanks!

Reply:
sure